samza-core/src/main/scala/org/apache/samza/storage/ContainerStorageManager.java

```diff
@@ -43,53 +78,259 @@
+  }
+
+  /**
+   * Create taskStores with the given store mode.
+   * In case a store already exists in taskStores and it is a persistent store, the existing store is overwritten with a new one.
+   * Existing non-persistent stores in taskStores are left as-is.
+   *
+   */
+  private void createTaskStores(ContainerModel containerModel, JobContext jobContext, ContainerContext containerContext,
+      Map<String, StorageEngineFactory<Object, Object>> storageEngineFactories,
+      Map<String, SystemStream> changelogSystemStreams, Map<String, Serde<Object>> serdes,
+      Map<TaskName, TaskInstanceMetrics> taskInstanceMetrics,
+      Map<TaskName, TaskInstanceCollector> taskInstanceCollectors, StorageEngineFactory.StoreMode storeMode) {
+
+    // iterate over each task and each storeName
+    for (Map.Entry<TaskName, TaskModel> task : containerModel.getTasks().entrySet()) {
+      TaskName taskName = task.getKey();
+      TaskModel taskModel = task.getValue();
+
+      for (String storeName : storageEngineFactories.keySet()) {
+
+        if (this.taskStores.get(taskName) == null) {
+          this.taskStores.put(taskName, new HashMap<>());
+        }
+
+        // if this store has been already created in the taskStores, then re-create and overwrite it only if it is a persistentStore
+        if (this.taskStores.get(taskName).containsKey(storeName) && this.taskStores.get(taskName)
+            .get(storeName)
+            .getStoreProperties()
+            .isPersistedToDisk()) {
+
+          StorageEngine storageEngine =
+              createStore(storeName, taskName, taskModel, jobContext, containerContext, storageEngineFactories,
+                  changelogSystemStreams, serdes, taskInstanceMetrics, taskInstanceCollectors, storeMode);
+
+          // add created store to map
+          this.taskStores.get(taskName).put(storeName, storageEngine);
+
+          LOG.info("Re-created store {} for task {} because it a persistent store", storeName, taskName);
+        } else if (!this.taskStores.get(taskName).containsKey(storeName)) {
+          // if this store doesnt exist, then create it and add to taskStores
+
+          StorageEngine storageEngine =
+              createStore(storeName, taskName, taskModel, jobContext, containerContext, storageEngineFactories,
+                  changelogSystemStreams, serdes, taskInstanceMetrics, taskInstanceCollectors, storeMode);
+
+          // add created store to map
+          this.taskStores.get(taskName).put(storeName, storageEngine);
+
+          LOG.info("Created store {} for task {}", storeName, taskName);
+        } else {
+
+          LOG.info("Not re-creating store {} for task {} because it a non-persistent store", storeName, taskName);
+        }
+      }
+    }
+  }
+
+  /**
+   * Helper method to instantiate a StorageEngine with the given parameters.
+   */
+  private StorageEngine createStore(String storeName, TaskName taskName, TaskModel taskModel, JobContext jobContext,
```

Comment:
if they're helper methods, make them static - so that it's obvious that there's no dependency on any class field

Reply:
done -- made one static. The other was infact not a helper, so added comments on what it does.